source/_posts/durable-entities.md

```diff
@@ -0,0 +1,129 @@
+---
+title: Getting started with Durable Entities
+tags:
+  - Azure Functions
+  - Durable Entities
+  - Durable Functions
+  - Azure
+  - Functions
+  - Serverless
+  - Serverless Functions
+  - Actors
+  - Function Actors
+date: 2020-03-31 09:23:54
+---
+
+
+Entity functions define operations for reading and updating small pieces of state, known as durable entities. Like orchestrator functions, durable entities are functions with a special trigger type, the entity trigger. Unlike orchestrator functions, entity functions manage the state of an entity explicitly, rather than implicitly representing state via control flow. Entities provide a means for scaling out applications by distributing the work across many entities, each with a modestly sized state.
+
+Source: [Microsoft](https://docs.microsoft.com/nl-nl/azure/azure-functions/durable/durable-functions-entities?tabs=csharp)
+
+In this post, we will have a look at how you can get started with Durable Entities. We will go through the concepts that you need to understand and how to implement them. For this article, we will be building an application for a hospital to check if a hospital bed is still occupied or not. A big plus using Azure Durable Entities for this is that we don't need to worry about the plumbing. The framework abstracts most of the state persistence tasks for us.
+
+To get started with Durable Entities you need some basic knowledge about Azure functions and the concept of durable functions. A durable function uses a 3 part system. First, you have the Http client function that receives the external call. This function is always stateless because it just receives the request and validates it. After that, it creates an Orchestrator that handles the request flow. Ones the Orchestrator is created he can start to create Activities that do the actual work here. So just to quickly recap we have:
+
+- Http Client Function(Stateless) - Receives the request, validates it, Sets up the Orchestrator.
+- Orchestrator - Has the flow logic and triggers Activities.
+- Activities - This has the actual compute logic and does the work.
+
+For more information on these topics please see [here](https://docs.microsoft.com/nl-nl/azure/azure-functions).
+
+## What is an Entity?
+
+An Entity is what u expect it to be. If you worked with Entity-Framework or DDD before it should be a familiar term. An entity is a class consisting of properties that represents a set of data in some kind of storage.
+
+In our case, the state of the Entity is persisted in Azure Table Storage by the framework.  Entities in Durable Entities are a bit different in the sense that they are like small microservices (Actors) that have methods on them that you can call. In order to call an Entity we need to know its `Entity ID`. This Id uniquely identifies the Entity. An Entity-Id consists of a name and a key that together form a uniquely identifiable string.
```

Comment:
An Entity-Id consists of a name and a key that together

ik zie verder beneden dat name hier NameOf(BedEntity) is, is dat gebruikelijk om de entity naam te doen? want dan zou ik dat hierbij vermelden.

Reply:
Niet echt gebruikelijk je kunt hier natuurlijk alles van maken wat je wil als het maar unique is.